### PR TITLE
[RFC] vim-patch:7.4.707

### PR DESCRIPTION
**Pull request by buztard**
```
Problem:    Undo files can have their executable bit set.
Solution:   Strip of the executable bit. (Mikael Berthe)
```

https://github.com/vim/vim/commit/v7-4-707

Original patch:

``` diff
diff --git a/src/nvim/undo.c b/src/nvim/undo.c
--- a/src/nvim/undo.c
+++ b/src/nvim/undo.c
@@ -1614,8 +1614,8 @@ u_write_undo(name, forceit, buf, hash)
 #endif
     }

-    /* strip any s-bit */
-    perm = perm & 0777;
+    /* strip any s-bit and executable bit */
+    perm = perm & 0666;

     /* If the undo file already exists, verify that it actually is an undo
      * file, and delete it. */
diff --git a/src/nvim/version.c b/src/nvim/version.c
--- a/src/nvim/version.c
+++ b/src/nvim/version.c
@@ -742,6 +742,8 @@ static char *(features[]) =
 static int included_patches[] =
 {   /* Add new patch number below this line */
 /**/
+    707,
+/**/
     706,
 /**/
     705,
```

### CODE REVIEW

File: src/nvim/undo.c [hunk 1]
```diff
@@ -1035,8 +1035,8 @@ void u_write_undo(char_u *name, int forceit, buf_T *buf, char_u *hash)
     }
   }
 
-  /* strip any s-bit */
-  perm = perm & 0777;
+  /* strip any s-bit and executable bit */
```
Comment (glts) [inline src/nvim/undo.c]: `// Strip any sticky and executable bits.`?

**Comment by glts** [comment]
:+1:

**Comment by buztard** [comment]
Updated comment according style guide

**Comment by Pyrohh** [comment]
Huh, I can't seem to repro this. Here's the patch I apply to my `nvimrc`:

```diff
diff --git a/configs/.nvim/nvimrc b/configs/.nvim/nvimrc
index adad030..96459c0 100644
--- a/configs/.nvim/nvimrc
+++ b/configs/.nvim/nvimrc
@@ -60,6 +60,7 @@ Glaive codefmt plugin[mappings]
 let s:nvimdir=$HOME . '/.nvim'
 let &backupdir=s:nvimdir . '/backup'  " backup files
 let &directory=s:nvimdir . '/tmp'     " swap files
+let &undodir=s:nvimdir   . '/undo'    " undo files
 
 if !isdirectory(s:nvimdir)
   call mkdir(s:nvimdir, 'p')
@@ -73,6 +74,9 @@ if !isdirectory(&directory)
   call mkdir(&directory, 'p')
 endif
 
+if !isdirectory(&undodir)
+  call mkdir(&undodir, 'p')
+endif
 
 " Options
 set backspace=indent,eol,start
@@ -90,6 +94,7 @@ set nowrap
 set nrformats-=octal
 set numberwidth=3
 set shortmess+=asTI
+set undofile
 set virtualedit+=block
 " set lazyredraw
 
```

Any undo file which has had `chmod +x` promptly has the executable bit removed when it's accessed. Perhaps this only makes a difference for certain `umask` values?

**Comment by justinmk** [comment]
Merged. It may be system-specific. But it could not hurt.

**Pull request closed without merging**